kotlinx-coroutines-core/common/src/CoroutineScope.kt

```diff
@@ -117,13 +123,30 @@ public fun MainScope(): CoroutineScope = ContextScope(SupervisorJob() + Dispatch
 /**
  * Returns `true` when the current [Job] is still active (has not completed and was not cancelled yet).
  *
- * Check this property in long-running computation loops to support cancellation:
+ * A coroutine cancallation [is cooperative](https://kotlinlang.org/docs/cancellation-and-timeouts.html#cancellation-is-cooperative)
+ * and normally, the cancellation machinery happens when the coroutine *suspends*, for example,
```

Comment:
```suggestion
 * and normally, it's checked if a coroutine is cancelled when it *suspends*, for example,
```
I think that's more specific without being misleading.